core/libp2p/crypto/aes_provider/aes_provider_impl.hpp

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_LIBP2P_CRYPTO_IMPL_DETAIL_AES_CRYPT_IMPL_HPP
+#define KAGOME_CORE_LIBP2P_CRYPTO_IMPL_DETAIL_AES_CRYPT_IMPL_HPP
+
+#include "libp2p/crypto/aes_provider.hpp"
+
+#include <outcome/outcome.hpp>
+#include "common/buffer.hpp"
+#include "libp2p/crypto/common.hpp"
+
+namespace libp2p::crypto::aes {
+  class AesProviderImpl : public AesProvider {
+    using Buffer = kagome::common::Buffer;
+    using Aes128Secret = libp2p::crypto::common::Aes128Secret;
+    using Aes256Secret = libp2p::crypto::common::Aes256Secret;
+
+   public:
+    outcome::result<Buffer> encrypt_128_ctr(const Aes128Secret &secret,
```

Comment:
Why methods are not camelCase?

Reply:
Without underscore it looks ugly, and ptr is significant part of cipher modification name. Changing the order (ctr first, and then number looks incorrect).

Reply:
renamed methods as follows {encrypt, decrypt}AesCtr{128,256}, for example encryptAesCtr128